### PR TITLE
feat: add tekton pipeline run atf tests on pull request

### PR DESCRIPTION
**Pull request by ryankwilliams**
## Description

This PR adds the Tekton PipelineRun allowing new open PR's to be able to run tier 1 tests marked for PR check (shift left testing).

## Type of Change

- [x] CI / Tests 

## Self-Review Checklist

- [x] I have performed a self-review of my code
- [ ] I have added relevant comments to complex code sections
- [ ] I have updated documentation where needed
- [ ] I have considered the security impact of these changes
- [ ] I have considered performance implications
- [ ] I have thought about error handling and edge cases
- [ ] I have tested the changes in my local environment

## Summary by CodeRabbit

* **Chores**
  * Added an automated PR-triggered test pipeline to run ATF tests for pull requests, passing repo/PR/branch context and providing workspace storage for test artifacts.
  * Added Renovate configuration to enable automatic updates and automerge for Tekton catalog pipeline images.

### CODE REVIEW

**Comment by jay-steurer** [comment]
Hi @ryankwilliams  you have a few code rabbit issues. Can you address those or reply why they are not applicable? Thank you

File: renovate.json [hunk 1]
```diff
@@ -0,0 +1,5 @@
+{
+  "tekton": {
+    "automerge": true
+  }
```
[the remark this answers is not in the document]
Comment (ryankwilliams) [reply renovate.json]: The renovate config file has been revised in latest commit.

**Comment by ryankwilliams** [comment]
> Hi @ryankwilliams you have a few code rabbit issues. Can you address those or reply why they are not applicable? Thank you

Thank you @jay-steurer, I have revised the suggested change from coderabbit.

**Comment by jay-steurer** [comment]
@ryankwilliams  we are working on the CI issues, and we will get back at this PR shortly.

**Review by huffmanca — APPROVED** [review]

**Review by jeffh-oss — APPROVED** [review]
